app/src/main/java/io/neurolab/MathBasics.java

```diff
@@ -20,14 +20,10 @@ public static double getMedian(double[] input) {
     }
 
     public static double updateMean(double currentMean, int n, double val) {
+        if(n <= 0)
+            throw new IllegalArgumentException("Number of elements(n) cannot be less than or equal to 0");
         double nD = (double) n;
-        if (n > 1) {
-            double result = (double) ((1 / nD) * val);
-            result += (double) (((nD - 1) / (nD)) * currentMean);
-            return result;
-        }
-
-        return val;
+        return ( (nD - 1) * currentMean + val ) / nD;
```

Comment:
@RohitJain1103 I can see here that ` val ` was returned earlier. But here it is removed. Can you explain that?

Reply:
In the previous code val was returned as output when n <= 1 which is the same output in the new code if put n = 1. The formula on the last line is generalized formula.